### PR TITLE
Tools/px_uploader: exit if unsuitable board is connected

### PR DESCRIPTION
**Pull request by bkueng**
Avoids endless looping, and is mostly useful when used in automated upload scripts.

### CODE REVIEW

**Comment by LorenzMeier** [comment]
Who should review this?

File: Tools/px_uploader.py [hunk 1]
```diff
@@ -869,6 +873,11 @@ def main():
                     # print the error
                     print("\nERROR: %s" % ex.args)
 
+                except FirmwareNotSuitableException:
+                    unsuitable_board = True
+                    up.close()
+                    continue
```
Comment (julianoes) [inline Tools/px_uploader.py:879]: Why do you `continue` here and then check `unsuitable_board` instead of just doing `sys.exit(2)` right here?

Comment (bkueng) [reply Tools/px_uploader.py:879]: Because there could be other ports attached with a matching board id.

Comment (julianoes) [reply Tools/px_uploader.py:879]: Aha!

**Pull request merged**